File: app/src/api/sessions.js
import { request } from "./client.js";

// → { sessions: Session[] }
// Session = { job_id, title, source_type, source, status, archived, created_at,
//             duration_seconds, detected_language }
//   `source` is the canonical YouTube URL for source_type="youtube"
//   (e.g. https://www.youtube.com/watch?v=...), filename for "upload",
//   null for "transcript".
export const listSessions = () => request("/sessions");

File: gaz-server/src/routes/sessions.py
from fastapi import APIRouter

from ..db import jobs_repo
from ..schemas.jobs import SessionRow, SessionsResponse

router = APIRouter()


@router.get("/sessions", response_model=SessionsResponse)
async def list_sessions() -> SessionsResponse:
    rows = jobs_repo.list_sessions()
    return SessionsResponse(
        sessions=[
            SessionRow(
                job_id=r["id"],
                title=r.get("title"),
                source_type=r["source_type"],
                source=r.get("source"),
                status=r["status"],
                archived=bool(r.get("archived", False)),
                created_at=str(r["created_at"]),
                duration_seconds=r.get("duration_seconds"),
                detected_language=r.get("detected_language"),
            )
            for r in rows
        ]
    )
